packages/elements/src/utils/elementIs.ts

```diff
@@ -26,29 +32,45 @@ import type {
   SequenceMediaElement,
 } from '../types';
 
+function isMediaElement(e: Element): e is MediaElement;
+function isMediaElement(e: Draft<Element>): e is Draft<MediaElement>;
 function isMediaElement(e: Element): e is MediaElement {
-  return 'resource' in e;
+  return 'resource' in e && Boolean(e.resource);
 }
 
+function isTextElement(e: Element): e is TextElement;
+function isTextElement(e: Draft<Element>): e is Draft<TextElement>;
 function isTextElement(e: Element): e is TextElement {
-  return 'font' in e;
+  return 'font' in e && Boolean(e.font);
 }
 
+function isDefaultBackgroundElement(e: Element): e is DefaultBackgroundElement;
+function isDefaultBackgroundElement(
+  e: Draft<Element>
+): e is Draft<DefaultBackgroundElement>;
 function isDefaultBackgroundElement(e: Element): e is DefaultBackgroundElement {
-  return 'isDefaultBackground' in e;
+  return 'isDefaultBackground' in e && Boolean(e.isDefaultBackground);
 }
 
+function isBackgroundable(e: Element): e is BackgroundableElement;
+function isBackgroundable(e: Draft<Element>): e is Draft<BackgroundableElement>;
 function isBackgroundable(e: Element): e is BackgroundableElement {
   // All media is backgroundable.
-  return 'isBackground' in e || 'resource' in e;
+  return ('isBackground' in e && Boolean(e.isBackground)) || isMediaElement(e);
```

Comment:
`setBackgroundElement` tests are now failing, presumably because of this change?

Reply:
Fixed in 5fee6f4add.

Reply:
Fixed in 2fc423d

Reply:
Hm, couldn't `isBackground` be `false` though in theory for a backgroundable element? This currently checks now if it already is `isBackground`, not if it's backgroundable. Thoughts?

Reply:
@miina You mean it should be `typeof e.isBackground !== 'undefined'` instead of `Boolean(e.isBackground))`?

Reply:
Yep.